### DISCUSSION TITLE
samples\demo\dirbrowse crashed when running

### DISCUSSION THREAD

**Discussion by LukyGuyLucky** [aardappel/lobster#304] [Q&A] [answered]
Greetings!
Just 1 hour ago I downloaded the latest binaray package from here and setup in my [Win10 64 bit + VS community 2019],added the bin foler to system environment path and then try to run the sample:

:\lobster>bin\lobster samples\demos\dirbrowse.lobster
dirbrowse.lobster(108): warning: integer literal (2) where float expected

The appliocation crashed with a popup windows which says 
Caption: NVOGLDC invisible:lobster.exe fatal erro
Contnents: Please send E:\lobster\bin\lobster.exe.exp.log tot he developer!
samples\demos\dirbrowse.lobster

Was I missing something important,say depencies or something else that I don't have totally ?(Is so,what should I to setup a whole doable workspace to playaround all features in this language?)

Below is the logfile for your reference.
[lobster.exe.exp.log](https://github.com/aardappel/lobster/files/14467856/lobster.exe.exp.log)

Meanwhile,may I ask a couple of things please?

1. When my workspace is doable,I would like to learn with some tutorials from the beginning.Where should I get them please.

2. I tried another sample E:\lobster>bin\lobster samples\rosettacode\hello_world_test.lobster
it runs ,printed with "Goodbye, World!" output. So excited!
So where is the compiled bit-code or something alike

3. How to make those tests  in test folder compiles and running (test module)

4. And how can I compile and get an binaray executable file ,in windows ,that could be hello_world_test.exe (64 bit)

5 Does the language support building to static(lib)/dynamic(dll) output

6. How should I directly interop with c and/or c++, for example,simply link with ,say,sqlite3.lib to make use of this library

That simply means how to explore all features and try to use them by a  c/c++/d background junoir,maximum use this language to do some real practice,but NOT throw away c.

 Sorry for so long so navie questions.

I am really wanting to try this project.
Help would be apprecated.

Thanks and best regards,
Peter

**Comment by aardappel** [comment, marked as answer]
It's crashing deep in some nvidia dll:
`00007FFF83D16126 (nvwgf2umx): (filename not available): cask::ShaderList<cask::ConvShader,cask::Convolution>::availableShaders`

This appears to be an nvidia driver issue: https://www.nvidia.com/en-us/geforce/forums/game-ready-drivers/13/208329/windows-10-games-are-crashing-in-nvwgf2umxdll/

I see lots of similar crash reports online, like: https://forum.dcs.world/topic/273525-crash-when-entering-multiplayer/

Sounds like you might want to do a clean uninstall of the nvidia driver and install something newer / different?

**Reply by LukyGuyLucky** [reply]
Well note with thanks.

**Comment by aardappel** [comment]
As to your other questions:

1. Have you checked all the docs? There is a tutorial on how to write a game in there. There is no tutorial how to learn programming from scratch with Lobster, it is assumed you're familiar with some other language already.
2. By default it runs in JIT mode which executes directly and generates no executable. Generating an executable is a more complex process and involves using visual studio, see docs.
3. you can run the tests `bin/lobster tests/unittest.lobster`
4. Again, see docs.
5. No.
6. That is also in the docs.

**Reply by LukyGuyLucky** [reply]
Thanks again.
